src/main/java/io/cdap/plugin/gcp/firestore/source/QueryDocumentSnapshotToRecordTransformer.java

```diff
@@ -0,0 +1,223 @@
+          Instant dateInstant = ((Date) object).toInstant();
+          long micros = TimeUnit.SECONDS.toMicros(dateInstant.getEpochSecond());
+          return Math.addExact(micros, TimeUnit.NANOSECONDS.toMicros(dateInstant.getNano()));
+          /*
+        case DECIMAL:
+          ensureTypeValid(fieldName, object, Decimal128.class);
+          return extractDecimal(fieldName, (Decimal128) object, schema);
+          */
+        default:
+          throw new UnexpectedFormatException(String.format("Field '%s' is of unsupported type '%s'",
+            fieldName, fieldLogicalType.name().toLowerCase()));
+      }
+    }
+
+    Schema.Type fieldType = schema.getType();
+    switch (fieldType) {
+      case BOOLEAN:
+        ensureTypeValid(fieldName, object, Boolean.class);
+        return object;
+      case INT:
+        ensureTypeValid(fieldName, object, Integer.class);
+        return object;
+      case DOUBLE:
+        ensureTypeValid(fieldName, object, Double.class);
+        return object;
+      case BYTES:
+        /*
+        ensureTypeValid(fieldName, object, ObjectId.class, byte[].class);
+        if (object instanceof ObjectId) {
+          // ObjectId can be read as a bytes. Exactly 12 bytes are required
+          ByteBuffer buffer = ByteBuffer.allocate(12);
+          ObjectId objectId = (ObjectId) object;
+          objectId.putToByteBuffer(buffer);
+          buffer.rewind();
+          return buffer;
+        }
+        */
+        return object;
+      case LONG:
+        ensureTypeValid(fieldName, object, Long.class);
+        return object;
+      case STRING:
+        /*
+        ensureTypeValid(fieldName, object, ObjectId.class, String.class);
+        if (object instanceof ObjectId) {
+          // ObjectId can be read as a string
+          return object.toString();
+        }
+        */
+        return object;
+      /*
+      case MAP:
+        ensureTypeValid(fieldName, object, BSONObject.class);
+        return convertMap(fieldName, (BSONObject) object, schema);
+      case ARRAY:
+        ensureTypeValid(fieldName, object, BasicBSONList.class);
+        return convertArray(fieldName, (BasicBSONList) object, schema);
+      case RECORD:
+        ensureTypeValid(fieldName, object, BSONObject.class);
+        return convertRecord(fieldName, (BSONObject) object, schema);
+       */
+      default:
+        throw new UnexpectedFormatException(String.format("Field '%s' is of unsupported type '%s'", fieldName,
+          fieldType.name().toLowerCase()));
+    }
+  }
+
+  /*
+  private byte[] extractDecimal(String fieldName, Decimal128 decimal128, Schema schema) {
+    BigDecimal decimal = decimal128.bigDecimalValue();
+    int schemaPrecision = schema.getPrecision();
+    int schemaScale = schema.getScale();
+
+    if (decimal.precision() > schemaPrecision) {
+      throw new UnexpectedFormatException(
+        String.format("Field '%s' has precision '%s' which is higher than schema precision '%s'.",
```

Comment:
do we still need these and why are they commented out?

Reply:
Not required. Hence removed.